### DISCUSSION TITLE
Choose multiple picklist items in one field

### DISCUSSION THREAD

**Discussion by grantfitzsimmons** [specify/specify7#5589] [Feature Requests]
Picklists should have the ability to select multiple items if the user gives it the ability. Like you have 3 options: `A`, `B`, and `C`

You choose A, the field displays `A`
You choose B next, now the field displays `A, B`
Now you choose A again, the field displays `B`

This could work by creating new picklist items/using existing picklist items that are hidden from the user while allowing multiple values to be selected. 

**Example:**
A herbarium wants to select multiple locations. Currently, there must be dozens of picklist items rather than allowing the user to select the subset of herbaria one by one.

**Requested By:** New Mexico and others

**Comment by mc0822** [comment]
Zsuzsanna Papp of NHMD (Danish Natural History Museums) also requested this feature, via 9/6/22 email.

**Comment by maxpatiiuk** [comment]
I can't remember, did you say this was implemented in specify 6? How does it work (i.e, with querying, exports, search)

Does it store the same vale in PickListItem.value as in PickListItem.title?

**Comment by maxpatiiuk** [comment]
An ugly work around:
1. Create an agent formatter that formats group member agents using `;`
2. Create a new agent and set agent type to `Group`
3. Create agents `A` and `B` of type `Other` and add them to the group created in previous step
4. Create a pick list out of `Agent` table and use the formatter created in step 1
5. Use that pick list in the form for the desired field

Now, value like `A; B` would be available in the pick list.

Drawbacks:
- No way to restrict pick list items to just some Agents at the moment - it will be showing all agents, which is slow
- Misuses the Agent table - perhaps you can pick a different table for abuse (i.e, Preparation and PrepType)

**Comment by grantfitzsimmons** [comment]
>I hope you are doing well. I just have a quick question about the vitality field. Would it be possible to edit the vitality field to where we can select multiple options? Like one of those "select all that apply" drop downs. For some lots we would like to select both "alive" and "dead," or another combination of the five options.  

**Requested By:** Chris at the Bailey-Matthews National Shell Museum & Aquarium

**Comment by grantfitzsimmons** [comment]
>I can't remember, did you say this was implemented in specify 6? How does it work (i.e, with querying, exports, search)

This is not implemented in Specify 6

**Comment by lexiclevenger** [comment]
This feature has also been requested by Corinna P at CSIRO via Asana.
